streaming/consumer.cc

```diff
@@ -59,8 +59,8 @@ std::function<future<> (flat_mutation_reader_v2)> make_streaming_consumer(sstrin
                                              cf->get_sstables_manager().configure_writer(origin),
                                              encoding_stats{}, pc).then([sst] {
                     return sst->open_data();
-                }).then([cf, sst, offstrategy, reason] {
-                    if (offstrategy && (reason == stream_reason::repair)) {
+                }).then([cf, sst, offstrategy, reason, origin] {
+                    if (offstrategy && sstables::repair_origin == origin) {
```

Comment:
There's a problem. Starting offstrategy after a table is repaired means compaction competes with the next table's repair.

Although, both tasks have to be done.

Don't you have to remove another offstrategy call from somewhere else? i.e. move the offstrategy trigger from somewhere else to here? Where is it now?

Reply:
1. agree
2. ok